internal/interfaceadapter/repository/userrepo/repository.go

```diff
@@ -24,6 +30,178 @@ func New(connManager ConnManager) *repository {
 	}
 }
 
+// TODO: こいつをどこに置くかは後でなやむ
+func GetBusinesses() map[uint64]string {
+	return map[uint64]string{
+		0:   "選択してください",
+		1:   "広告代理店",
+		2:   "飲食業界",
+		3:   "テレビ業界",
+		4:   "芸能関係",
+		5:   "マスコミ",
+		6:   "商社",
+		7:   "外資系",
+		8:   "航空",
+		9:   "IT業界",
+		101: "その他",
+	}
+}
+
+// TODO: こいつをどこに置くかは後でなやむ
+func GetJobs() map[uint64]string {
+	return map[uint64]string{
+		0:   "選択してください",
+		1:   "社長",
+		2:   "役員",
+		3:   "部長",
+		4:   "OL",
+		5:   "秘書",
+		6:   "シェフ",
+		7:   "パティシエ",
+		8:   "バーテンダー",
+		9:   "芸能人/著名人",
+		10:  "フードアナリスト",
+		11:  "フードコーディネーター",
+		12:  "グルメブロガー",
+		13:  "ソムリエ",
+		14:  "デザイナー",
+		15:  "広報",
+		16:  "ライター",
+		17:  "大学生",
+		18:  "ビジネスマン",
+		101: "その他",
+	}
+}
+
+func (r *repository) GetCategoryTypeMap(ctx context.Context, lang string) (map[uint64]string, error) {
+	query := `
+SELECT category_type, category_name
+FROM category_type_name_translation
+WHERE lang = ?
+`
+	rows, err := r.connManager.GetSlave().QueryContext(ctx, query, lang)
+	if err != nil {
+		return nil, fmt.Errorf("db access error: %w", err)
+	}
+	defer rows.Close()
+
+	categoryTypes := make(map[uint64]string)
+	for rows.Next() {
+		var (
+			categoryType uint64
+			categoryName string
+		)
+
+		err := rows.Scan(&categoryType, &categoryName)
+		if err != nil {
+			return nil, err
+		}
+
+		categoryTypes[categoryType] = categoryName
+	}
+	return categoryTypes, nil
+}
+
+func (r *repository) GetProfileByUserIDs(ctx context.Context, ids []uint64) (map[uint64]*entity.UserProfile, error) {
+	queryTemplate := `
+SELECT 
+user_id, 
+company, company_privacy, 
+business, business_privacy, 
+job, job_privacy, 
+organization, organization_privacy, 
+category_types, category_types_privacy
+FROM user_profile
+WHERE user_id IN (%s)
+`
+	args := make([]interface{}, 0, len(ids))
+	for _, id := range ids {
+		args = append(args, id)
+	}
+	placeholder := strings.Join(strings.Split(strings.Repeat("?", len(ids)), ""), ",")
+	query := fmt.Sprintf(queryTemplate, placeholder)
+	rows, err := r.connManager.GetSlave().QueryContext(ctx, query, args...)
+	if err != nil {
+		return nil, fmt.Errorf("db access error: %w", err)
+	}
+	defer rows.Close()
+
+	businessNameMap := GetBusinesses()
+	jobNameMap := GetJobs()
+	categoryTypeMap, err := r.GetCategoryTypeMap(ctx, CategoryTypeLangJapanese)
+	if err != nil {
+		return nil, fmt.Errorf("db access error: %w", err)
```

Comment:
`db access error`だと汎用的すぎてどこの処理のエラーかわかりにくいので、実行しようとした処理がわかるような情報があるともっと良いですね
```suggestion
		return nil, fmt.Errorf("get category type map: %w", err)
```